### DISCUSSION TITLE
Disable highlighting globally in console.print()

### DISCUSSION THREAD

**Discussion by scottb5612** [Textualize/rich#3433] [Q&A]
Is there a way to disable automatic highlighting globally,
to avoid the need to include "highlight=False" on each call to console.print()?

**Comment by willmcgugan** [comment]
Second paragraph.

https://rich.readthedocs.io/en/latest/highlighting.html#highlighting

**Comment by scottb5612** [comment]
Is there a way to achieve the effect with the print method alone?

E.g. the following highlights the email address

print("[b red]Hi![/] [italic yellow] python", "tree.py ***@***.***>")

**Reply by TomJGooding** [reply]
Could you try clarifying your question? Do you mean you are still seeing highlighting after disabling everywhere in the console?

**Comment by scottb5612** [comment]
Surely.
In this program, I see the following output.
What I want is to disable highlight for the call to print() at line 2  as
well
(and  for any other calls to   print())

console = Console(highlight=False)
print("[b red]Hi![/] [italic yellow] python", "tree.py ***@***.***>")
console.print("[b red]Hi![/] [italic yellow] python", "tree.py ***@***.***
>")

[image: Screenshot 2024-07-18 at 8.53.47 PM.png]

**Reply by TomJGooding** [reply]
Sorry but I can't see the screenshot you posted.

But if you're talking about Python's print function, Rich obviously has no control over that. 

If you're seeing highlighting just using the plain print function, that's likely applied by your terminal.

**Comment by scottb5612** [comment]
Since rich is intercepting print(), and adding highlighighting, isn't there
a global function
that could be defined via a function provided by rich?  I had tried using
the constructor
in the hope that this might work, but no it didn't

console = Console(highlight=False)

The output appeared as follows for the program (I changed the font color in
the mail
so you can see tha catual output

Here is the code:

rom rich import print
from rich.console import Console
console = Console(highlight=False)
highlight=False
print("[b red]Hi![/] ***@***.***>")
console.print("[b red]Hi![/] ***@***.***>")

Here is the output

*Hi!* ***@***.***>
*Hi!* ***@***.***>

**Reply by willmcgugan** [reply]
```python
from rich import reconfigure
reconfigure(highlight=False)
```

**Comment by scottb5612** [comment]
Got it.. thanks.
